server/src/main/java/org/apache/druid/server/coordination/ServerManager.java

```diff
@@ -293,21 +302,28 @@ public ServerManager(
         queryMetrics -> queryMetrics.segment(segmentIdString)
     );
 
-    CachingQueryRunner<T> cachingQueryRunner = new CachingQueryRunner<>(
-        segmentIdString,
-        segmentDescriptor,
-        objectMapper,
-        cache,
-        toolChest,
-        metricsEmittingQueryRunnerInner,
-        cachePopulator,
-        cacheConfig
-    );
+    QueryRunner<T> queryRunner;
```

Comment:
`CachingQueryRunner` already has the machinery to decide whether or not to use the caches, should we just pass the optional in and use it as another input to the existing `useCache`/`populateCache` equation so we don't need these if statements in `ServerManager` and `SinkQuerySegmentWalker`?

Reply:
I was thinking that less places worry about the different meanings of `cacheKeyPrefix` content, the better it is. So for CachingQueryRunner, `cacheKeyPrefix` is just a plain dumb byte array